superset-frontend/src/datasource/ChangeDatasourceModal.tsx

```diff
@@ -135,54 +163,112 @@ const ChangeDatasourceModal: FunctionComponent<ChangeDatasourceModalProps> = ({
   const changeSearch = (
     event: React.FormEvent<FormControl & FormControlProps>,
   ) => {
-    setFilter((event.currentTarget?.value as string) ?? '');
+    const searchValue = (event.currentTarget?.value as string) ?? '';
+    setFilter(searchValue);
   };
 
-  const data = useMemo(
-    () =>
-      filter && datasources
-        ? datasources.filter((datasource: any) =>
-            TABLE_FILTERABLE.some(field => datasource[field]?.includes(filter)),
-          )
-        : datasources,
-    [datasources, filter],
-  );
+  const handleChangeConfirm = () => {
+    SupersetClient.get({
+      endpoint: `/datasource/get/${confirmedDataset.type}/${confirmedDataset.id}`,
+    })
+      .then(({ json }) => {
+        onDatasourceSave(json);
+        onChange(`${confirmedDataset.id}__table`);
+      })
+      .catch(response => {
+        getClientErrorObject(response).then(
+          ({ error, message }: { error: any; message: string }) => {
+            const errorMessage = error
+              ? error.error || error.statusText || error
+              : message;
+            addDangerToast(errorMessage);
+          },
+        );
+      });
+    onHide();
+    addSuccessToast('Successfully changed datasource!');
+  };
+
+  const handlerCancelConfirm = () => {
+    setConfirmChange(false);
+  };
+
+  const renderTableView = () => {
+    const data = resourceCollection.map((ds: any) => ({
+      rawName: ds.table_name,
+      connection: ds.database.database_name,
+      schema: ds.schema,
+      name: (
+        <a
+          href="#"
+          onClick={() => selectDatasource({ type: 'table', ...ds })}
+          className="datasource-link"
+        >
+          {ds.table_name}
+        </a>
+      ),
+      type: ds.kind,
+    }));
+
+    return data;
+  };
 
   return (
-    <Modal
+    <StyledModal
       show={show}
       onHide={onHide}
       responsive
       title={t('Select a dataset')}
       hideFooter
     >
       <>
-        <Alert bsStyle="warning">
-          <strong>{t('Warning!')}</strong> {CHANGE_WARNING_MSG}
-        </Alert>
-        <div>
-          <FormControl
-            inputRef={ref => {
-              setSearchRef(ref);
-            }}
-            type="text"
-            bsSize="sm"
-            value={filter}
-            placeholder={t('Search / Filter')}
-            onChange={changeSearch}
-          />
-        </div>
-        {loading && <Loading />}
-        {datasources && (
-          <TableView
-            columns={TABLE_COLUMNS}
-            data={data}
-            pageSize={20}
-            className="table-condensed"
-          />
+        {!confirmChange && (
+          <>
+            <Alert bsStyle="warning">
```

Comment:
I like the warning message here, I should use it in the overwrite confirmation dialog!